app/src/features/surveys/components/SamplingMethodForm.tsx

```diff
@@ -44,7 +44,7 @@ const SamplingMethodForm = () => {
 
   const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, index: number) => {
     setAnchorEl(event.currentTarget);
-    setEditData({ ...values.methods[index], index });
```

Comment:
This is the root issue basically. The index needs to be tracked, but originally it was being added into the methods data object, which then meant it was sent to the API.

Now, we still track it, but in a separate property of a wrapping object.